test/e2e/memtierd.test-suite/memtierd.source.sh

```diff
@@ -62,13 +62,47 @@ memtierd-meme-start() {
         command-error "failed to start meme, pid not found"
     fi
     if [[ -n "$MEME_CGROUP" ]]; then
-        vm-command "mkdir /sys/fs/cgroup/$MEME_CGROUP; echo $MEME_PID > /sys/fs/cgroup/$MEME_CGROUP/cgroup.procs"
-        if [[ -n "$MEME_MEMS" ]]; then
-            vm-command "echo ${MEME_MEMS} > /sys/fs/cgroup/$MEME_CGROUP/cpuset.mems"
-        fi
+        vm-command "mkdir /sys/fs/cgroup/$MEME_CGROUP; echo \"${MEME_MEMS}\" > /sys/fs/cgroup/$MEME_CGROUP/cpuset.mems; echo $MEME_PID > /sys/fs/cgroup/$MEME_CGROUP/cgroup.procs"
     fi
 }
 
 memtierd-meme-stop() {
     vm-command "killall -KILL meme"
+    if [[ -n "$MEME_CGROUP" ]]; then
+        vm-command "sudo rmdir /sys/fs/cgroup/$MEME_CGROUP"
+    fi
+}
+
+next-round() {
+    local round_counter_var=$1
+    local round_counter_val=${!1}
+    local round_counter_max=$2
+    local round_delay=$3
+    if [[ "$round_counter_val" -ge "$round_counter_max" ]]; then
+        return 1
+    fi
+    eval "$round_counter_var=$(($round_counter_val + 1))"
+    sleep $round_delay
+    return 0
+}
+
+memtierd-verify-scanned-pids() {
+    local pid_count=$#
+
+    for pid_regexp in "$@"; do
+      round_number=0
+      while ! ( memtierd-command "stats -t memory_scans -f csv | awk -F, \"{print \\\$1}\""; grep ${pid_regexp} <<< $COMMAND_OUTPUT) ; do
+        echo "grep pid matching ${pid_regexp} not found"
+        next-round round_number 5 1 || {
+            error "timeout: memtierd did not watch ${pid_regexp}"
+        }
+      done
+    done
+
+    echo "The expected count of scanned pids is ${pid_count}"
+    # in case pid_count is 0, so calculate the output of memory_scans table from the 3rd line and increase pid_count by 1
+    pid_count=$(($pid_count+1))
+    if ! ( memtierd-command "stats -t memory_scans -f csv | awk -F, \"NR>=3 {count++} END {print count}\""; grep ${pid_count} <<< $COMMAND_OUTPUT); then
```

Comment:
Test development tip: insert a new line `interactive`  as line 105 and run the test. You will get a prompt on this line. Then you can run in the prompt:

```
run.sh> memtierd-command 'stats -t memory_scans -f csv'
```

and observe COMMAND_OUTPUT with all its lined numbered:

```
run.sh> echo "$COMMAND_OUTPUT" | nl -ba
     1  stats -t memory_scans -f csv
     2  table: memory scans
     3  pid,scans,tot[pages],avg[s],last,max[s],avg[G],last[G],a+w[%],last
     4
     5  13755,134,24687447,0.076317,0.127130,0.182618,0.702800,0.714314,0.086983 0.082776
     6  13770,134,24794243,0.051390,0.102332,0.146920,0.705840,0.722424,0.086367 0.081846
     7  13786,134,24914929,0.068302,0.122974,0.168487,0.709275,0.722057,0.086318 0.081888
     8  memtierd>
```

And to see what is the number returned by the awk, I can try in the same interactive prompt:

```
run.sh> memtierd-command 'stats -t memory_scans -f csv | awk -F, "NR>=3 {count++} END {print count}"'
...
run.sh> echo "$COMMAND_OUTPUT" | nl -ba
     1  stats -t memory_scans -f csv | awk -F, "NR>=3 {count++} END {print count}"
     2  4
     3  memtierd>
```

That is, $COMMAND_OUTPUT contains also the command that is executed in memtierd prompt. Therefore number "3" can be grep'd from $COMMAND_OUTPUT as the command contains "NR>=3". As a result, this implementation passes fine even if I give it only two pids as a parameter, because it (above) increased pid_count by 1 to 3, and then grep matched NR>=3, yet the number of real pids returned by memtierd would be something different.

I'd suggest simplifying line 105 to something like

```
memtierd-command 'stats -t memory_scans -f csv'
observed_pid_count="$(grep ^[0-9] <<< "$COMMAND_OUTPUT" | wc -l)"
if [[ "$observed_pid_count"  != "$expected_pid_count" ]]; then
    error "expected $expected_pid_count pids, but got $observed_pid_count"
fi
```